projects/packages/videopress/src/client/block-editor/hooks/use-video-data/index.ts

```diff
@@ -72,7 +77,8 @@ export default function useVideoData( {
 				setIsRequestingVideoData( false );
 
 				// pick filename from the source_url
-				const filename = response.original?.split( '/' )?.at( -1 );
+				const filename =
```

Comment:
Debugging the reasons this code [wasn't working with Android](https://github.com/Automattic/jetpack/pull/29997#pullrequestreview-1379015953) led me to this line. Although I couldn't find any open issues that specifically mention that Hermes doesn't work with the `at()` method, I did find it isn't supported by [some older browsers](https://caniuse.com/mdn-javascript_builtins_array_at), and replacing it fixes the issue with Android. I therefore strongly suspect the cause of the issues was Hermes not supporting the `at()` method.

Another way of rewriting this could be: `const filename = response.original?.split('/').slice(-1)[0];` but I thought that the use of `length` may be more easily parsed, although it's more verbose. I'd be interested to hear other thoughts or opinions on this.

Reply:
> Debugging the reasons this code https://github.com/Automattic/jetpack/pull/29997#pullrequestreview-1379015953 led me to this line. Although I couldn't find any open issues that specifically mention that Hermes doesn't work with the at() method, I did find it isn't supported by [some older browsers](https://caniuse.com/mdn-javascript_builtins_array_at), and replacing it fixes the issue with Android. I therefore strongly suspect the cause of the issues was Hermes not supporting the at() method.

Great investigation! It's true that there's no reference to the `at` function in [the Hermes language features](https://hermesengine.dev/docs/language-features/), so as you mentioned might be an incompatibility with the WebView Android version ([it's supported from `92` onwards](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility)). However, I checked my Android WebView version and it's `111`, so not sure where's the incompatibility coming from 🤔.

>Another way of rewriting this could be: const filename = response.original?.split('/').slice(-1)[0]; but I thought that the use of length may be more easily parsed, although it's more verbose. I'd be interested to hear other thoughts or opinions on this.

Another approach would be to import a polyfill for the function. I checked the [`at` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#see_also) and there's a [polyfill in the `core-js` library that we could use](https://github.com/zloirock/core-js#relative-indexing-method). Technically, we could import it into Gutenberg by adding the following code to [`globals.js `](https://github.com/WordPress/gutenberg/blob/73fb8a63c59b98208c599405ae3071d9b88417b9/packages/react-native-editor/src/globals.js#L10-L13):
```
import 'core-js/proposals/relative-indexing-method';
```

WDYT?

Reply:
I've just tested my above suggestion on an actual Android device and works 🎊.

Reply:
@SiobhyB I took the liberty of opening a PR to include the polyfill in Gutenberg. Let me know if you could take a look and test this PR with the polyfill. Thanks!

Reply:
Oh, brilliant, thanks for that! Approved the Gutenberg PR and reverted my change to the code in https://github.com/Automattic/jetpack/pull/29997/commits/bfc9f37215428521c8d539cb357ef88d8bd289bd. I'll also go ahead to update the Gutenberg submodule in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5652.

Reply:
> Oh, brilliant, thanks for that! Approved the Gutenberg PR and reverted my change to the code in [bfc9f37](https://github.com/Automattic/jetpack/commit/bfc9f37215428521c8d539cb357ef88d8bd289bd). I'll also go ahead to update the Gutenberg submodule in [wordpress-mobile/gutenberg-mobile#5652](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5652) once it's merged.

Great, I've just merged [the PR](https://github.com/WordPress/gutenberg/pull/49749).